link-grammar/api.c

```diff
@@ -537,7 +537,11 @@ void sentence_delete(Sentence sent)
 	pool_delete(sent->Exp_pool);
```

Comment:
above, you deleted what you're reusing!?

Reply:
As I couldn't find `pool_reuse(sent->Exp_pool)` in the code, I guess you referred to line 543:
`pool_reuse(sent->dict->Exp_pool)`.
But this is the dict Exp memory-pool.

BTW:
1. **All** the sentence memory pools are currently deleted at `sentence_delete()`.
But the dict pools survive - for a file dict they are just left untouched, and for a db dict they are reused.
2. I guess it is possible to save CPU by reusing memory-pools over sentences and deleting them on thread exit only.
This can be done by reusing them on `sentence_delete()` instead of deleting them, and deleting them using a registered exit function (I didn't test that so I don't know how much it would save).
A drawback is that if a single sentence happened to use a lot of memory, this memory would not get freed until exit.
I remedy could be adding an additional argument to `pool_reuse()` for how many memory-pool entries to retain.